### PR TITLE
Consolidate all the duplicated testdata, allow running from any directory

### PR DESCRIPTION
**Pull request by bradfitz**
- Unify all the ca.pem, server1.key, server1.pem files into one testdata
  directory.
- Add a function test.Abs() to return the absolute path of a file relative
  to the gRPC root. Use it from tests and examples.
- Now examples can be run from any directory and still find their TLS
  or JSON input files.

### CODE REVIEW

**Comment by LetsUseGerrit** [comment]
Gerrit code review: https://go-review.googlesource.com/19436 (at git rev 60f2c4e)

File: test/end2end_test.go [hunk 1]
```diff
@@ -31,7 +31,7 @@
  *
  */
 
-package grpc_test
+package test
```
Comment (tamird) [inline test/end2end_test.go]: While you're here, would you mind moving this file up to the repo root and leaving it in `package grpc_test`? The current makeup of `package test` is confusing, where's it's half test utilities and half real tests.

**Comment by bradfitz** [comment]
@iamqizhao, are you ready for this one if I update it to tip?

**Comment by dfawley** [comment]
@bradfitz if you are still interested in fixing this, we'd love to have it.  Thanks!

**Comment by dfawley** [comment]
#1297 is a newer attempt to fix this.

**Pull request closed without merging**